test/plog/LogStreamTest.cpp

```diff
@@ -0,0 +1,449 @@
+
+        // let start() finish and then run the tests
+        _testTimer.set_callback([this] {
+            _testFuture = seastar::make_ready_future()
+            .then([this] {
+                // Create a persistence cluster
+                dto::PersistenceGroup group1{.name="Group1", .plogServerEndpoints = _plogConfigEps()};
+                dto::PersistenceCluster cluster1;
+                cluster1.name="Persistence_Cluster_1";
+                cluster1.persistenceGroupVector.push_back(group1);
+
+                auto request = dto::PersistenceClusterCreateRequest{.cluster=std::move(cluster1)};
+                return RPC().callRPC<dto::PersistenceClusterCreateRequest, dto::PersistenceClusterCreateResponse>(dto::Verbs::CPO_PERSISTENCE_CLUSTER_CREATE, request, *_cpoEndpoint, 1s);
+            })
+            .then([this] (auto&& response) {
+                // Init and create a logstream
+                auto& [status, resp] = response;
+                K2EXPECT(log::ltest, status, Statuses::S201_Created);
+
+                ConfigVar<String> configEp("cpo_url");
+                String cpo_url = configEp();
+
+                return _mmgr.init(cpo_url, "Partition-1", "Persistence_Cluster_1", false);
+            })
+            .then([this] { return runTest1();})
+            .then([this] { return runTest2();})
+            .then([this] { return runTest3();})
+            .then([this] {
+                K2LOG_I(log::ltest, "======= All tests passed ========");
+                exitcode = 0;
+            })
+            .handle_exception([this](auto exc) {
+                try {
+                    std::rethrow_exception(exc);
+                } catch (RPCDispatcher::RequestTimeoutException& exc) {
+                    K2LOG_E(log::ltest, "======= Test failed due to timeout ========");
+                    exitcode = -1;
+                } catch (std::exception& e) {
+                    K2LOG_E(log::ltest, "======= Test failed with exception [{}] ========", e.what());
+                    exitcode = -1;
+                }
+            })
+            .finally([this] {
+                K2LOG_I(log::ltest, "======= Test ended ========");
+                seastar::engine().exit(exitcode);
+            });
+        });
+        _testTimer.arm(0ms);
+        return seastar::make_ready_future<>();
+    }
+
+    seastar::future<> runTest1() {
+        K2LOG_I(log::ltest, ">>> Test1: Write and Read a Value from/to one log stream");
+        _logStream = _mmgr.obtainLogStream(LogStreamType::WAL);
+        String header("0", 10000);
+        Payload payload([] { return Binary(20000); });
+        payload.write(header);
+        // write a string to the first log stream
+        return _logStream->append(std::move(payload))
+        .then([this, header] (auto&& response){
+            auto& [status, plogId, appended_offset] = response;
+            if (!status.is2xxOK()){
+                throw std::runtime_error("cannot append data!");
```

Comment:
use K2ASSERT here to assert the status is 200

Reply:
fixed